### PR TITLE
Add support for password aging on Solaris

### LINKED ISSUE #1280: Feature request: expand functionality of "user" module
##### ISSUE TYPE

Feature Idea
##### COMPONENT NAME

user module
##### ANSIBLE VERSION

N/A
##### SUMMARY

The user module allows us to modify aspects of a user's account, including the password and some controls on the password.  This is apparently done (primarily) through the useradd, usermod, and userdel operating system commands.  The passwd command on most Linux distros and on Solaris (and perhaps other Unix variants) allows use of the -n, -x, and -w options to control password aging.  I would like to see those capabilities added to the user module.  I'm not proficient in Python and even less familiar with the inner workings of Ansible modules, otherwise I would add password aging capabilities myself.  My immediate interest is actually in disabling password aging (passwd -x -1 username, for example) but on some systems I'll need to enable it and set the parameters.

### PR DESCRIPTION
**Pull request by bsdlme**
- Feature Pull Request
##### COMPONENT NAME

system/user.py
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /usr/local/etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

```
##### SUMMARY

The user module was missing support for password aging (min days, max days and warn days). This patch adds the missing functionality.

Fixes #1280 

```

```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @larsengels for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by bsdlme** [comment]
@gregdek What should I do here? The checks only fail with Python 3.x and I did not add the failing code snippet.

**Comment by gregdek** [comment]
Hmm, looks like this is a process issue. Marking with bot_broken for now and we'll review.

**Comment by gregdek** [comment]
Putting this into core_review, since this module is owned by the core team. @larsengels thanks for your patience with this one.

**Comment by gregdek** [comment]
Thanks @larsengels for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by bsdlme** [comment]
@gregdek No problem. I'm using this now with as override module, so I can already use it.

**Comment by bsdlme** [comment]
Oh, and by the way: I think the shadow file should never get directly edited. It should get copied to a temporary file, that one should be edited and copied back.

**Comment by bcoca** [comment]
@larsengels that is why we have the atomic_move function, ansible should always be working on temp files.

File: system/user.py [hunk 1]
```diff
@@ -1388,12 +1419,17 @@ def modify_user_usermod(self):
...
+                            fields[4] = str(int(maxweeks) * 7)
+                        if warnweeks:
+                            fields[5] = str(int(warnweeks) * 7)
                         line = ':'.join(fields)
                         lines.append('%s\n' % line)
                     open(self.SHADOWFILE, 'w+').writelines(lines)
                     rc = 0
-                except Exception:
-                    err = get_exception()
+                except Exception, err:
```
Comment (abadger) [inline system/user.py]: @larsengels This change is the source of the python3 failure you're getting.

Comment (bsdlme) [inline system/user.py]: Oops. Thank you! This is now fixed in e902976

**Comment by bsdlme** [comment]
@bcoca Thank you for pointing me at atomic_move(). When I find the time I will implement it and send another PR.

**Comment by gregdek** [comment]
@larsengels A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by bsdlme** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @larsengels. To the current maintainers, @grossws please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by bsdlme** [comment]
Ping!

**Pull request merged**